### ISSUE TITLE
[Bug]: Latest CoreCLR Plugin does not work

### ISSUE THREAD

**Issue by ertup0** [yukieiji/UnityExplorer#1]
### Are you on the latest version of UnityExplorer?

- [X] Yes, I'm on the latest version of UnityExplorer.

### Which release are you using?

BepInEx IL2CPP

### Which game did this occur on?

PowerWash Simulator
Unity 2020.3.38f1

### Describe the issue.

The game works fine with bepinex but the plugin does not load
Same error and log output on storyteller game

### Relevant log output

```shell
[Message: Preloader] BepInEx 6.0.0-be.667 - PowerWashSimulator
[Message: Preloader] Built from commit 6b500b35727716f8ba182dedae304a628dbbb608
[Info   :   BepInEx] System platform: Windows 10 64-bit
[Info   :   BepInEx] Process bitness: 64-bit (x64)
[Info   :   BepInEx] Running under Unity 2020.3.38f1
[Info   :   BepInEx] Runtime version: 6.0.7
[Info   :   BepInEx] Runtime information: .NET 6.0.7
[Info   : Preloader] 0 patcher plugins loaded
[Info   : Preloader] 0 assemblies discovered
[Message:AssemblyPatcher] Executing 0 patch(es)
[Message:   BepInEx] Chainloader initialized
[Info   :Il2CppInterop] Registered mono type Il2CppInterop.Runtime.DelegateSupport+Il2CppToMonoDelegateReference in il2cpp domain
[Info   :   BepInEx] 1 plugin to load
[Info   :   BepInEx] Loading [UnityExplorer 4.9.1]
[Message:UnityExplorer] UnityExplorer 4.9.1 initializing...
[Message:UnityExplorer] [UniverseLib] UniverseLib 1.5.1 initializing...
[Info   :Il2CppInterop] Registered mono type UniverseLib.UniversalBehaviour in il2cpp domain
[Info   :Il2CppInterop] Registered mono type UniverseLib.Runtime.Il2Cpp.Il2CppManagedEnumerator in il2cpp domain
[Message:UnityExplorer] [UniverseLib] Finished UniverseLib initial setup.
[Info   :Il2CppInterop] Registered mono type UnityExplorer.ExplorerBehaviour in il2cpp domain
[Message:UnityExplorer] Initialized UnityCrashPrevention for: Canvas.get_renderingDisplaySize
[Message:   BepInEx] Chainloader startup complete
[Message:UnityExplorer] [UniverseLib] Loaded Unhollowed modules in 3,7410002 seconds.
[Message:UnityExplorer] [UniverseLib] Setup deobfuscation cache in 0,22700001 seconds.
[Message:UnityExplorer] [UniverseLib] Initialized Legacy Input support
[Message:UnityExplorer] [UniverseLib] 	 IL2CPP method has no corresponding pointer, aborting patch of UnityEngine.AssetBundle.UnloadAllAssetBundles
[Info   :Il2CppInterop] Registered mono type UniverseLib.AssetBundle in il2cpp domain
[Warning:UnityExplorer] [UniverseLib] Exception parsing Unity version, falling back to old AssetBundle load method...
[Error  :Il2CppInterop] Exception in IL2CPP-to-Managed trampoline, not passing it to il2cpp: System.TypeInitializationException: The type initializer for 'Il2CppInterop.Runtime.InteropTypes.Arrays.Il2CppStructArray`1' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'Il2CppInterop.Runtime.Il2CppClassPointerStore`1' threw an exception.
 ---> System.InvalidOperationException: Sequence contains more than one matching element
   at System.Linq.ThrowHelper.ThrowMoreThanOneMatchException()
   at System.Linq.Enumerable.TryGetSingle[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at Il2CppInterop.Runtime.Il2CppClassPointerStore`1..cctor() in /home/runner/work/Il2CppInterop/Il2CppInterop/Il2CppInterop.Runtime/Il2CppClassPointerStore.cs:line 67
   --- End of inner exception stack trace ---
   at Il2CppInterop.Runtime.InteropTypes.Arrays.Il2CppArrayBase`1.StaticCtorBody(Type ownType) in /home/runner/work/Il2CppInterop/Il2CppInterop/Il2CppInterop.Runtime/InteropTypes/Arrays/Il2CppArrayBase.cs:line 83
   at Il2CppInterop.Runtime.InteropTypes.Arrays.Il2CppStructArray`1..cctor() in /home/runner/work/Il2CppInterop/Il2CppInterop/Il2CppInterop.Runtime/InteropTypes/Arrays/Il2CppStructArray.cs:line 9
   --- End of inner exception stack trace ---
   at Il2CppInterop.Runtime.InteropTypes.Arrays.Il2CppStructArray`1.op_Implicit(T[] arr) in /home/runner/work/Il2CppInterop/Il2CppInterop/Il2CppInterop.Runtime/InteropTypes/Arrays/Il2CppStructArray.cs:line 48
   at UniverseLib.AssetBundle.LoadFromMemory(Byte[] binary, UInt32 crc)
   at UniverseLib.UI.UniversalUI.LoadBundle(String id)
   at UniverseLib.UI.UniversalUI.LoadBundle()
   at UniverseLib.UI.UniversalUI.Init()
   at UniverseLib.Universe.SetupCoroutine()+MoveNext()
   at UniverseLib.Runtime.Il2Cpp.Il2CppManagedEnumerator.MoveNext()
   at Trampoline_ByteThisUniverseLib.Runtime.Il2Cpp.Il2CppManagedEnumeratorMoveNext(IntPtr , Il2CppMethodInfo* )
```

**Comment by ertup0** [comment]
this is the one that I used. From my testing it gives this error in all games with unity 2020-2021

**Comment by ertup0** [comment]
1. for some reason my Install doesn't have mono folder. unhollowed and unity-libs are empty
2. I installed among us today 35 mins ago just to test the plugin so there are no mods in it
3. among us and all my games are installed in the default steam directory "C:\Program Files (x86)\Steam\steamapps\common\Among Us"

**Comment by ertup0** [comment]
1. ok i made a mistake unhollowed and unity libs are not empty i'm sorry

**Comment by yukieiji** [comment]
In 6.0.0-be667, the unhollowed folder is never created, so I think this issue is probably caused by installing a new BepInEx with the old BepInEx still in place.

**Comment by yukieiji** [comment]
So if you delete the BepInEx folder and install BepInEx 6.0.0-be.667 and start it, and then install UnityExplorer, it should work.

**Comment by ertup0** [comment]
But the problem is I never installed other bepinex versions on among us. I downloaded clean version of the game today from steam and then installed 667 version there are no leftovers from old versions

**Comment by yukieiji** [comment]
I think that the initialization of BepInEx was not completed correctly because a large number of files were created during the initialization process, which may have been blocked by security or other factors, or because the game was force-quit in the middle of the initialization process.

**Comment by ertup0** [comment]
okay i just deleted all bepinex folders and reinstalled the game then I downloaded the 667 bepinex build and UnityExplorer.BepInEx.Unity.IL2CPP.CoreCLR plugin, disabled Windows defender and I still get the same error log

**Comment by yukieiji** [comment]
Do other publicly available mods work properly?

**Comment by yukieiji** [comment]
Here is the build that I tried to fixed the issue that was similar to the UniverseLib issues.
Is it fixed now?
[UnityExplorer.BepInEx.Unity.IL2CPP.CoreCLR.zip](https://github.com/yukieiji/UnityExplorer/files/11672175/UnityExplorer.BepInEx.Unity.IL2CPP.CoreCLR.zip)

**Comment by yukieiji** [comment]
fixed in [v4.9.2](https://github.com/yukieiji/UnityExplorer/releases/tag/v4.9.2)
